src/engine/SpatialJoin.cpp

```diff
@@ -230,6 +230,15 @@ size_t SpatialJoin::getCostEstimate() {
 // ____________________________________________________________________________
 uint64_t SpatialJoin::getSizeEstimateBeforeLimit() {
   if (childLeft_ && childRight_) {
+    // If we limit the number of results to k, even in the largest scenario, the
+    // result can be at most `|childLeft| * k`
+    auto maxResults = getMaxResults();
+    if (maxResults.has_value()) {
+      return childLeft_->getSizeEstimate() * maxResults.value();
+    }
+
+    // If we don't limit the number of results, we cannot draw conclusions about
+    // the size, other than the worst case `|childLeft| * |childRight|`
```

Comment:
Yes, but assuming the worst case is also a little pessimistic and probably also leads to bad plans.
What do ypu thiing about adding a constant like `SPATIAL_MAX_DISTANCE_JOIN_COST_ESTIMATE` factor 
(e.g. 0.001) and then return `left * right * factor`. (Add this constant to `Constants.h` and document it).

Reply:
Thanks. I think this is a very reasonable idea, as it improves the results in practice while maintaining the correct asymptotic behavior. :+1: